compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -2269,9 +2269,12 @@ public void visit(BLangRecordVariableDef bLangRecordVariableDef) {
     public void visit(BLangErrorVariable bLangErrorVariable) {
         analyzeNode(bLangErrorVariable.typeNode, env);
         populateUnusedVariableMapForNonSimpleBindingPatternVariables(this.unusedLocalVariables, bLangErrorVariable);
-        this.currDependentSymbolDeque.push(bLangErrorVariable.symbol);
-        analyzeNode(bLangErrorVariable.expr, env);
-        this.currDependentSymbolDeque.pop();
+        // When we visit this path from on fail clause the symbol is null
```

Comment:
Is this true for capture binding patterns?

Reply:
for `on fail error msg` kind of scenarios the flow doesn't go into this method right